op-chain-ops/crossdomain/withdrawal.go

```diff
@@ -0,0 +1,116 @@
+package crossdomain
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+var _ WithdrawalMessage = (*Withdrawal)(nil)
+
+// Withdrawal represents a withdrawal transaction on L2
+type Withdrawal struct {
+	Nonce    *big.Int
+	Sender   *common.Address
+	Target   *common.Address
+	Value    *big.Int
+	GasLimit *big.Int
+	Data     []byte
+}
+
+// NewWithdrawal will create a Withdrawal
+func NewWithdrawal(
+	nonce *big.Int,
+	sender, target *common.Address,
+	value, gasLimit *big.Int,
+	data []byte,
+) *Withdrawal {
+	return &Withdrawal{
+		Nonce:    nonce,
+		Sender:   sender,
+		Target:   target,
+		Value:    value,
+		GasLimit: gasLimit,
+		Data:     data,
+	}
+}
+
+// Encode will serialize the Withdrawal so that it is suitable for hashing.
+func (w *Withdrawal) Encode() ([]byte, error) {
+	args := abi.Arguments{
+		{Name: "nonce", Type: Uint256Type},
+		{Name: "sender", Type: AddressType},
+		{Name: "target", Type: AddressType},
+		{Name: "value", Type: Uint256Type},
+		{Name: "gasLimit", Type: Uint256Type},
+		{Name: "data", Type: BytesType},
+	}
+	enc, err := args.Pack(w.Nonce, w.Sender, w.Target, w.Value, w.GasLimit, w.Data)
+	if err != nil {
+		return nil, err
+	}
+	return enc, nil
+}
+
+// Decode will deserialize a Withdrawal
+func (w *Withdrawal) Decode(data []byte) error {
+	args := abi.Arguments{
+		{Name: "nonce", Type: Uint256Type},
+		{Name: "sender", Type: AddressType},
+		{Name: "target", Type: AddressType},
+		{Name: "value", Type: Uint256Type},
+		{Name: "gasLimit", Type: Uint256Type},
+		{Name: "data", Type: BytesType},
+	}
+	decoded, err := args.Unpack(data)
+	if err != nil {
+		return err
+	}
+
+	sender, ok := decoded[1].(common.Address)
+	if !ok {
```

Comment:
Why are sender/target done with the `ok` syntax (no panics), but the rest are done a in a manner that could panic (if the type was not correct in the result of `args.Unpack`).

Reply:
Ok. I agree that this is very unlikely to cause a problem, I was curious why there was a difference.